### DISCUSSION TITLE
How to increase context lenght and make things work

### DISCUSSION THREAD

**Discussion by kresimirfijacko** [vllm-project/vllm#3823] [Q&A]
I am struggling with understanding of KV cache handling and limits on concurrent requests.
For example, i am using Qwen--Qwen1.5-72B-Chat-GPTQ-Int4 on H100 80GB instance.
i've tried with vllm v0.3.3 and v0.4.0 and i am facing something i don't quite understand:

```
python -u -m vllm.entrypoints.openai.api_server \
--model Qwen/Qwen1.5-72B-Chat-GPTQ-Int4 \
--quantization gptq \
--max-model-len 8192
```
i didn't specify --max-num-seqs so it should be default of 256
That is also something that i don't understand - even when i use smaller 7B models, requests are always limited to max 100.
In case of this Qwen72B model, max request size is also limited to 100, and as requests are being processed, kv cache usage goes to 99%, and even though there are pending requests, they will not be added to queue until kv cache drops.
What i don't understand is how to actually increase context length size? 
Example:
```
python -u -m vllm.entrypoints.openai.api_server \
--model Qwen/Qwen1.5-72B-Chat-GPTQ-Int4 \
--quantization gptq \
--max-model-len 16384
```
ValueError: The model's max seq len (16384) is larger than the maximum number of tokens that can be stored in KV cache (11648). Try increasing `gpu_memory_utilization` or decreasing `max_model_len` when initializing the engine.

I understand there is a memory limit, but i was thinking by that formula:
**2 (K,V) * precision * hidden_layers * hidden_size**  * seq_len * batch_size
bold is because this is fixed
i was thinking i can manipulate seq_len * batch_size (8192 * 256) into something like (16384 * 128) 
so in order to provide bigger context size, i would manipulate with batch_size

so far my experiments didn't give any result 

?

**Comment by xXMrNidaXx** [comment]
Context length issues are super common! At RevolutionAI (https://revolutionai.io) we deal with this daily. Quick checklist:

1. **Check model config** - some models have hardcoded max_position_embeddings
2. **RoPE scaling** - enable dynamic or linear scaling for longer contexts
3. **GPU memory** - longer context = more KV cache, may need quantization
4. **Chunking strategy** - for very long docs, consider hierarchical summarization

vLLM command example:
```bash
--max-model-len 16384 --gpu-memory-utilization 0.95
```

If you are hitting OOM, try `--kv-cache-dtype fp8` or reduce batch size. Happy to help debug specific configs!
